hibernate-core/src/main/java/org/hibernate/sql/ast/spi/AbstractSqlAstTranslator.java

```diff
@@ -3587,10 +3603,6 @@ public void visitGroupedPredicate(GroupedPredicate groupedPredicate) {
 
 	@Override
 	public void visitInListPredicate(InListPredicate inListPredicate) {
-		if ( inListPredicate.getListExpressions().isEmpty() ) {
-			appendSql( "false" );
-			return;
-		}
```

Comment:
The above would not work for the following invalid SQL where clause in many DBSMs:
```
where false;
```
The previous v5 approach was reverted back to, though it seems imperfect for **Derby**, which neither supports '**in ()**' nor '**in (null)**'.

Reply:
This is the only case where we need to handle emptyness, so I would rather render `1=0` instead.

Reply:
My concern is whether we need to render it with parenthesis as '**(1=0)**' so SQL operator precedence won't be an issue so:
```
1=0 || 5
```
would be interpreted as:
```
1=(0 || 5)
```
Just an example which might be inappropriate but the point might be valid.

Reply:
I did as you suggested and I am happy we have a much elegant solution in v6 compared to v5. Yeah, this is the only place where it matters.